src/projections.jl

```diff
@@ -152,6 +158,117 @@ function vec_symm(X)
     return X[LinearAlgebra.triu(trues(size(X)))]
 end
 
+"""
+    projection_on_set(::DefaultDistance, v::AbstractVector{T}, ::MOI.ExponentialCone) where {T}
+
+projection of vector `v` on closure of the exponential cone
+i.e. `cl(Kexp) = {(x,y,z) | y e^(x/y) <= z, y>0 } U {(x,y,z)| x <= 0, y = 0, z >= 0}`.
+
+References:
+* [Proximal Algorithms, 6.3.4](https://web.stanford.edu/~boyd/papers/pdf/prox_algs.pdf)
+by Neal Parikh and Stephen Boyd.
+* [Projection, presolve in MOSEK: exponential, and power cones]
+(https://docs.mosek.com/slides/2018/ismp2018/ismp-friberg.pdf) by Henrik Friberg
+"""
+function projection_on_set(::DefaultDistance, v::AbstractVector{T}, ::MOI.ExponentialCone) where {T}
+    # https://web.stanford.edu/~boyd/papers/pdf/prox_algs.pdf, Section 6.3
+    length(v) != 3 && throw(DimensionMismatch("Mismatch between value and set"))
+
```

Comment:
you can replace the if elseif elseif... blocks by:
```julia
if something
    return
end
if something else
    return
end
...
```

Reply:
What's the advantage of this approach? Performance?

Reply:
Not performance, but basically staying on a low indentation.
Reading the code of a long function with a "return and forget" lets you just focus on the present line without remembering previous branching

Reply:
Changed